File: sb-ecom/src/main/kotlin/dev/alimansour/sbecom/model/Order.kt
package dev.alimansour.sbecom.model

import jakarta.persistence.*
import java.time.LocalDate

@Entity
@Table(name = "orders")
class Order(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Column(nullable = false)
    var email: String,

    @OneToMany(
        mappedBy = "order",
        cascade = [CascadeType.PERSIST, CascadeType.MERGE],
        orphanRemoval = true
    )
    var orderItems: MutableList<OrderItem> = mutableListOf(),

    var orderDate: LocalDate,

    @OneToOne
    @JoinColumn(name = "payment_id")
    var payment: Payment,

    var totalAmount: Double,
    var orderStatus: String,

    @ManyToOne
    @JoinColumn(name = "address_id")
    var address: Address?
)

File: ecom-frontend/src/types/Order.ts
import {OrderItem} from "./OrderItem";

export interface Order {
    id: number;
    email: string;
    orderItems: OrderItem[];
    orderDate: string;
    payment?: {
        id: number | string;
        paymentMethod: string;
        pgPaymentId: string;
        pgStatus: string;
        pgResponseMessage: string;
        pgName: string;
    };
    totalAmount: number;
    orderStatus: string;
    addressId?: number | string;
}